src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -1100,6 +1100,17 @@ private void logProvisionInfo(String message) {
         RunInstancesRequest riRequest = entry.getKey();
         List<Filter> diFilters = entry.getValue();
 
+        // Ensure our describe instances request instance does not filter on a specific subnet id
```

Comment:
I think it would be more efficient to remove the `subnet-id` filter altogether from here
https://github.com/mkozell/ec2-plugin/blob/dfedd02b6546c9476c4792e7756e760d21f62161/src/main/java/hudson/plugins/ec2/SlaveTemplate.java#L1008

Reply:
What about the callers to the `makeRunInstancesRequestAndFilters` method that want/need the `subnet-id` included in the List of Filters that are returned from this method?

For example, if a caller, calls the `getDescribeInstanceResult` method with the `allSubnets` set to `false`, then this caller wants `subnet-id` to be included in the filter.  If we remove Line 1008 as you are suggesting, we will be changing the current implementation of `getDescribeInstanceResult`.